### PR TITLE
feat(activation): split application activation

### PR DESCRIPTION
**Pull request by Phil91**
## Description

Split application activation process step into multiple process steps

## Why

To have the possibility to retrigger each process by itself

## Issue

Refs: #913

## Checklist

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/docs/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [x] I have performed [IP checks](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-04#checking-libraries-using-the-eclipse-dash-license-tool) for added or updated 3rd party libraries
- [x] I have created and linked IP issues or requested their creation by a committer
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [x] I have added tests that prove my changes work
- [x] I have checked that new and existing tests pass locally with my changes
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/registration/ApplicationActivation.Library/ApplicationActivationService.cs [hunk 1]
```diff
@@ -241,24 +232,138 @@ private async Task RemoveRegistrationRoles(Guid applicationId, IUserRolesReposit
...
+        return new IApplicationChecklistService.WorkerChecklistProcessStepExecutionResult(
+            ProcessStepStatusId.DONE,
+            entry =>
+            {
+                entry.ApplicationChecklistEntryStatusId = ApplicationChecklistEntryStatusId.DONE;
+            },
+            null,
+            null,
+            true,
+            failedUserNames.Count == 0 ? null : $"user(s) {string.Join(",", failedUserNames)} had no assigned email");
```
Comment (ntruchsess) [inline src/registration/ApplicationActivation.Library/ApplicationActivationService.cs:360]:
```suggestion
        var failedUserNames = await PostRegistrationWelcomeEmailAsync(context.ApplicationId, companyName, businessPartnerNumber, cancellationToken).ToListAsync().ConfigureAwait(false);

        return new IApplicationChecklistService.WorkerChecklistProcessStepExecutionResult(
            ProcessStepStatusId.DONE,
            entry =>
            {
                entry.ApplicationChecklistEntryStatusId = ApplicationChecklistEntryStatusId.DONE;
            },
            null,
            null,
            true,
            failedUserNames.Count == 0
                ? null
                : $"user(s) {string.Join(",", failedUserNames)} had no assigned email");

```

File: src/registration/ApplicationActivation.Library/ApplicationActivationService.cs [hunk 2]
```diff
@@ -241,24 +232,138 @@ private async Task RemoveRegistrationRoles(Guid applicationId, IUserRolesReposit
...
+    }
+
+    private async Task<List<string>> PostRegistrationWelcomeEmailAsync(Guid applicationId, string companyName, string businessPartnerNumber, CancellationToken cancellationToken)
```
Comment (ntruchsess) [inline src/registration/ApplicationActivation.Library/ApplicationActivationService.cs:363]:
```suggestion
    private async IAsyncEnumerable<string> PostRegistrationWelcomeEmailAsync(Guid applicationId, string companyName, string businessPartnerNumber, [EnumeratorCancellation] CancellationToken cancellationToken)
    {
        var failedUserNames = new List<string>();
        await foreach (var user in portalRepositories.GetInstance<IApplicationRepository>().GetEmailDataUntrackedAsync(applicationId).WithCancellation(cancellationToken).ConfigureAwait(false))
        {
            var userName = string.Join(" ", new[] { user.FirstName, user.LastName }.Where(item => !string.IsNullOrWhiteSpace(item)));
            if (string.IsNullOrWhiteSpace(user.Email))
            {
                yield return userName;
                continue;
            }

            var mailParameters = ImmutableDictionary.CreateRange(new[]
            {
                KeyValuePair.Create("userName", !string.IsNullOrWhiteSpace(userName) ? userName : user.Email),
                KeyValuePair.Create("companyName", companyName),
                KeyValuePair.Create("bpn", businessPartnerNumber),
                KeyValuePair.Create("homeUrl", _settings.PortalHomeAddress),
                KeyValuePair.Create("passwordResendUrl", _settings.PasswordResendAddress),
                KeyValuePair.Create("companyRolesParticipantUrl", _settings.CompanyRolesParticipantAddress),
                KeyValuePair.Create("dataspaceUrl", _settings.DataspaceAddress)
            });
            mailingProcessCreation.CreateMailProcess(user.Email, "EmailRegistrationWelcomeTemplate", mailParameters);
        }
    }
```

**Review by ntruchsess — CHANGES_REQUESTED** [review]
see comments

**Comment by Phil91** [comment]
@ntruchsess fixed the pr findings :)

**Review by ntruchsess — APPROVED** [review]

**Review by ntruchsess — APPROVED** [review]

**Pull request merged**